### PR TITLE
Add the IISMiddleware via a IStartupFilter.

### PR DESCRIPTION
**Pull request by Tratcher**
#95 @DamianEdwards @davidfowl 

I'll submit a separate PR to Hosting to make ConfigureServices callable multiple times.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.IISIntegration/IISAddressExtensions.cs [hunk 1]
```diff
@@ -16,7 +18,7 @@ public static class IISAddressExtensions
         /// </summary>
         /// <param name="app"></param>
         /// <returns></returns>
-        public static IWebHostBuilder UseIISUrl(this IWebHostBuilder app)
+        public static IWebHostBuilder UseIIS(this IWebHostBuilder app)
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.IISIntegration/IISAddressExtensions.cs]: pass `Func<string, string>` and use that in the tests.

**Comment by Tratcher** [comment]
Updated. Environment variables have been abstracted. I've also removed the IApplicationBuilder extensions, we only use the IWebHostBuilder extension now.

File: test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs [hunk 2]
```diff
@@ -75,18 +106,19 @@ public class IISMiddlewareTests
...
+                .ConfigureServices(services =>
                 {
-                    Environment.SetEnvironmentVariable("ASPNETCORE_TOKEN", "TestToken");
-                    app.Use((context, next) =>
-                    {
-                        context.Request.Headers["MS-ASPNETCORE-TOKEN"] = "TestToken";
-                        return next();
-                    });
-                    app.UseIIS(new IISOptions
+                    services.AddSingleton(new IISOptions()
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs]: What? Shouldn't this be configure options?

File: src/Microsoft.AspNetCore.Server.IISIntegration/IISWebHostExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        /// <param name="app"></param>
+        /// <returns></returns>
+        public static IWebHostBuilder UseIIS(this IWebHostBuilder app)
+        {
+            if (app == null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            var port = app.GetSetting(ServerPort);
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.IISIntegration/IISWebHostExtensions.cs]: Nice and testable

File: test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs [hunk 4]
```diff
@@ -33,25 +37,50 @@ public class IISMiddlewareTests
...
+        }
+
+        [Fact]
+        public async Task AddsAuthenticationHandlerByDefault()
+        {
+            var assertsExecuted = false;
+
+            var builder = new WebHostBuilder()
+                .UseSetting("TOKEN", "TestToken")
+                .UseSetting("PORT", "12345")
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs]: Nice

File: test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs (hunk 2 above)
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs]: Oh, is that how that's supposed to work? Updated.

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**